### PR TITLE
CI: Read-only cache usage for main-native + jackson

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -133,6 +134,7 @@ jobs:
       - name: Jackson compatibility tests / 2.6.4 for Spark 2.4.4
         uses: gradle/gradle-build-action@v2
         with:
+          cache-read-only: true
```
Comment (XN137) [inline .github/workflows/main.yml:137]: seems unintentional that all other jackson versions below still write to the cache?

**Review by XN137 — COMMENTED** [review]
whats the rationale ?
too large cache size currently ?

File: .github/workflows/main.yml (hunk 1 above)
Comment (snazy) [reply .github/workflows/main.yml:137]: Only the first usage of the gradle-action pulls the cache

**Comment by snazy** [comment]
> whats the rationale ?

otherwise (without this change) the native + jackson jobs also push their new cache - which conflicts with the cache pushed by the "main" job.

File: .github/workflows/main.yml (hunk 1 above)
Comment (XN137) [reply .github/workflows/main.yml:137]: ok, adding it would be redundant but maybe also more consistent?

**Review by XN137 — APPROVED** [review]

**Pull request merged**